RealmSwift/RealmCollection.swift

```diff
@@ -84,9 +84,16 @@ public protocol _RealmMapValue {
     /// Advance to the next element and return it, or `nil` if no next element exists.
     public mutating func next() -> Element? {
         let next = generatorBase.next()
-        if let key = next as? Key,
-           let value = collection[key as AnyObject].map(dynamicBridgeCast) as? Value {
-            return (key: key, value: value)
+        if let key = next as? Key {
+            if let value = collection[key as AnyObject].map(dynamicBridgeCast) as? Value {
```

Comment:
This is going to perform very poorly. Reading a List element is going to allocate and discard two extra List objects, along with do a very large number of dynamic casts which are pretty expensive in Swift.